instrumentation/openai-client-instrumentation/src/main/java/co/elastic/otel/openai/wrappers/InstrumentedChatCompletionService.java

```diff
@@ -0,0 +1,271 @@
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.otel.openai.wrappers;
+
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.ERROR_TYPE;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_OPENAI_REQUEST_RESPONSE_FORMAT;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_OPENAI_REQUEST_SEED;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_OPERATION_NAME;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_REQUEST_FREQUENCY_PENALTY;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_REQUEST_MAX_TOKENS;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_REQUEST_MODEL;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_REQUEST_PRESENCE_PENALTY;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_REQUEST_STOP_SEQUENCES;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_REQUEST_TEMPERATURE;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_REQUEST_TOP_P;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_RESPONSE_FINISH_REASONS;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_RESPONSE_ID;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_RESPONSE_MODEL;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_SYSTEM;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_USAGE_INPUT_TOKENS;
+import static co.elastic.otel.openai.wrappers.GenAiAttributes.GEN_AI_USAGE_OUTPUT_TOKENS;
+
+import com.google.errorprone.annotations.MustBeClosed;
+import com.openai.core.RequestOptions;
+import com.openai.core.http.StreamResponse;
+import com.openai.models.ChatCompletion;
+import com.openai.models.ChatCompletionChunk;
+import com.openai.models.ChatCompletionCreateParams;
+import com.openai.models.CompletionUsage;
+import com.openai.services.blocking.chat.CompletionService;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.jetbrains.annotations.NotNull;
```

Comment:
same as previously, should we `@Nullable` instead ?

Reply:
Removed